### ISSUE TITLE
Cannot make it work with Z20 Pro

### ISSUE THREAD

**Issue by albaintor** [wizmo2/zidoo-player#18]
Hi,

I can successfully add the device after entering the IP address, it detects the model (Z20 Pro), but that's all : I can't control the device nor access to the library
I have tried to enable or disable authentication with no luck.
Is it a compatibility issue with the new models (2023) or something wrong on my side ?
Thanks

**Comment by wizmo2** [comment]
Can you enable debugging in the Integration settings and see what is in the logs
![image](https://github.com/wizmo2/zidoo-player/assets/11602094/79c92799-820e-48b4-961c-00eeda847aa6)

**Comment by albaintor** [comment]
Hi, I have enabled logs and I get this error message :
Logger: custom_components.zidoo.zidoorc
Source: custom_components/zidoo/zidoorc.py:448
Integration: Zidoo (documentation, issues)
First occurred: 6 janvier 2024 à 16:04:17 (19983 occurrences)
Last logged: 10:34:22

Exception: get() takes 1 positional argument but 2 were given

**Comment by wizmo2** [comment]
That's good, but I need a little more information if possible 

Could you restart HA, then (after a few minutes) download the log file (SETTINGS/SYSTEM/LOGS/Download Full Log.  

I need to see
- the initialization response (debug message)
- the full error message (so I can see what command is creating the problem).

For example:
`2024-01-07 10:15:07.592 DEBUG (SyncWorker_3) [custom_components.zidoo.zidoorc] connected: {'status': 200, 'model': 'Z9X', 'i..`

**Comment by albaintor** [comment]
Hi again, here it is
[home-assistant_2024-01-07T12-08-37.861Z.log](https://github.com/wizmo2/zidoo-player/files/13853354/home-assistant_2024-01-07T12-08-37.861Z.log)

Thanks

**Comment by wizmo2** [comment]
As you say, there is no additional information here.  The error is from the standard python `requests` library, so I don't think it is a zidoo issue.   It may be something to do with your home-assistant setup (`requests` still works in the platform I test with, but HA are pushing migration to the async verison `aiohttp`)

Can you confirm you used only the ip address when setting up the player (e.g. 192.168.1.20 not http://192.168.1.20 and not 192.168.20:80) 

Can you let me know 
- HA version
- HA hardware details

It will be a couple of days, but I'll look at adding a beta version with some more diagnostics.

**Comment by albaintor** [comment]
I am using the latest version : 2024.1.2 but I had this issue with previous updates too.
I have a Odroid N2+ setup with homeassistant blue.
Thanks for investigating

**Comment by albaintor** [comment]
Hi again, I think I have (I am not sure) found the problem after reviewing your code and reproducing it through postman.
All the requests work including the initial post that you make as followed (at the end of this post)
The problem is that you expect a cookie in return, but in my case the Zidoo won't return any cookie and it is not necessary to make other calls work : if it returns a 200 code, it is powered on .

I found another API to check if a media is running : /ZidooVideoPlay/getPlayStatus => returns
{"status":806,"msg":"The resource does not exist"} if nothing is playing
OR
 200 with the details of the media being played like in this example :
[{
    "status": 200,
    "video": {
        "status": 1,
        "title": "Hunger Games",
        "path": "/data/system/smb/192.168.1.10#Videos/Science-fiction/The.Hunger.Games.2012.MULTi.TRUEFRENCH.2160p.UHD.BluRay.REMUX.DoVi.HEVC-Zapax.mkv",
        "currentPosition": 880263,
        "duration": 8571000,
        "width": 3840,
        "height": 2160,
        "fps": 23.97599983215332,
        "formt": "H.265",
        "filesize": "73,06 GiB",
        "bitrate": "85,0 Mb/s",
        "output": "3840x2160P @ 60Hz  YUV420  10 Bits  ITU 709  ",
        "subtitleInfo": "French(PGS Forcés)-Forced PGS (1/2)",
        "audioInfo": "French DTS-HD 7.1 channel 48,0KHz (1/3)"
    },
    "subtitle": {
        "index": 1,
        "information": "French(PGS Forcés)-Forced PGS (1/2)"
    },
    "audio": {
        "index": 0,
        "information": "French DTS-HD 7.1 channel 48,0KHz (1/3)"
    },
    "playMode": {
        "index": 0,
        "information": "Désactivé"
    },
    "zoom": {
        "index": 1,
        "information": "Ajustement"
    },
    "3D": {
        "index": 0,
        "information": "2D",
        "isMvc3D": false
    }
}](url)



```
curl --location 'http://192.168.1.40:9529/ZidooControlCenter/getModel ' \
--header 'Connection: keep-alive' \
--header 'Content-Type: application/json' \
--data '{
    "method": "actRegister",
    "params": [
        {"clientid": "guest", "nickname": "Client", "level": "private"},
        [{"value": "yes", "function": "WOL"}]
    ],
    "id": 1,
    "version": "1.0"
}'
```

**Comment by albaintor** [comment]
Nevermind, this is not the cause : the problem comes from the requests.get which cause an "Exception: get() takes 1 positional argument but 2 were given". I am not a Python expert, it is unclear what causes this error

**Comment by albaintor** [comment]
I found the fix : the params argument needed to be named.
Just add "params=" and it works now
response = requests.get(
                    url, **params=**params, cookies=self._cookies, timeout=timeout, headers=headers
                )

**Comment by albaintor** [comment]
I just made a pull request. I let you review it. Thank you

**Comment by wizmo2** [comment]
Excellent.  Thank you very much.  Been travelling this week, was going to look at it this weekend.

**Comment by wizmo2** [comment]
Should be fixed in v1.4.2 with #19.  Thanks @albaintor
